identity-service/src/notifications/processNotifications/reactionNotification.js

```diff
@@ -0,0 +1,50 @@
+const models = require('../../models')
+const { notificationTypes } = require('../constants')
+
+async function processReactionNotifications (notifications, tx) {
+  for (const notification of notifications) {
+    const { slot, initiator: reactorId, metadata: { reaction_value: reactionValue, reacted_to: reactedTo, reaction_type: reactionType } } = notification
+
+    // First find the originating tip receive notif
+    const receiveTipNotification = await models.SolanaNotification.findOne({
+      where: {
+        type: notificationTypes.TipReceive,
+        metadata: {
+          [models.Sequelize.Op.contains]: {
+            'tipTxSignature': reactedTo
+          }
+        }
+      }
+    })
+
+    if (!receiveTipNotification) {
+      console.error(`Received reaction to ${reactedTo} from ${reactorId}, but not corresponding tip notification`)
+      return notifications
+    }
+
+    // Set it's reaction value
+    receiveTipNotification.metadata = { ...receiveTipNotification.metadata, reactionValue }
+
+    // Save out reaction value on receive tip, and create new reaction notif
+    await Promise.all([
```

Comment:
prob fine but are there any implications for one of these failing?

Reply:
great q. because we give each of these db ops the same transaction object, if any of them fail they all should - which is what we'd like, to avoid inconsistent states.